From d23bf3f70104933a7b36bae09983987762e623f2 Mon Sep 17 00:00:00 2001
From: Jason King <jason.brian.king@gmail.com>
Date: Thu, 13 Jul 2017 16:59:58 +0000
Subject: [PATCH] OS-6231 smartos-live should use existing copy of custr
 OS-6232 smart-os live src/Makefile doesn't pass DESTDIR correct to subdirs

---
 src/Makefile                                 |  3 ++-
 src/common/strings/Makefile                  |  3 ++-
 {tools/common => src/common/strings}/custr.c |  0
 {tools/common => src/common/strings}/custr.h |  4 +++-
 src/common/strings/strpath.c                 |  5 ++---
 src/common/strings/strpath.h                 |  4 ++--
 src/common/strings/tests/testpath.c          |  4 ++--
 src/dockerinit/Makefile                      |  8 ++++----
 src/dockerinit/json-nvlist/json-nvlist.c     |  3 ++-
 src/dockerinit/json-nvlist/json-nvlist.h     |  3 +--
 src/dockerinit/src/docker-common.h           |  4 ++--
 src/dockerinit/src/dockerinit.c              |  1 +
 src/routeinfo/Makefile                       | 13 ++++++++++---
 src/routeinfo/main.c                         |  3 +--
 tools/tzcheck/main.c                         |  2 ++
 15 files changed, 36 insertions(+), 24 deletions(-)
 rename {tools/common => src/common/strings}/custr.c (100%)
 rename {tools/common => src/common/strings}/custr.h (97%)

diff --git a/src/Makefile b/src/Makefile
index 050daaee..ab156b33 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -39,7 +39,8 @@ JSSTYLE_OLDSKOOL_OPTS =
 # without extra effort; it is overridden when make is invoked in this
 # subdirectory.
 #
-DESTDIR =		../proto
+# This should be an absolute path, or building $(SUBDIRS) will fail
+DESTDIR =		$(realpath ../proto)
 
 #
 # DESTNAME is used for the manifest target.  We set it to something
diff --git a/src/common/strings/Makefile b/src/common/strings/Makefile
index e8078229..2a198197 100644
--- a/src/common/strings/Makefile
+++ b/src/common/strings/Makefile
@@ -44,6 +44,7 @@ TEST_PROGRAMS = \
 TEST_BINS =	$(TEST_PROGRAMS:%=obj/%)
 
 STRINGS_OBJECTS = \
+		custr.o \
 		strlist.o \
 		strpath.o
 
@@ -56,7 +57,7 @@ CTFMERGE =	/usr/bin/true
 CC =		$(GCC)
 CSTYLE =	../../../tools/cstyle
 
-LIBS =		-lcmdutils -lumem
+LIBS =		-lumem
 
 COMPILE_C =	$(CC) $(CPPFLAGS) $(LDFLAGS) $(CFLAGS) -c -o $@ $< $(LIBS)
 
diff --git a/tools/common/custr.c b/src/common/strings/custr.c
similarity index 100%
rename from tools/common/custr.c
rename to src/common/strings/custr.c
diff --git a/tools/common/custr.h b/src/common/strings/custr.h
similarity index 97%
rename from tools/common/custr.h
rename to src/common/strings/custr.h
index f3433951..447f2a74 100644
--- a/tools/common/custr.h
+++ b/src/common/strings/custr.h
@@ -10,7 +10,7 @@
  */
 
 /*
- * Copyright 2015 Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 #ifndef _CUSTR_H
@@ -20,6 +20,8 @@
  * Private copy of custr*() utilities from libcmdutils.
  */
 
+#include <sys/types.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
diff --git a/src/common/strings/strpath.c b/src/common/strings/strpath.c
index 68cf4c85..c7cace75 100644
--- a/src/common/strings/strpath.c
+++ b/src/common/strings/strpath.c
@@ -10,14 +10,13 @@
  */
 
 /*
- * Copyright 2015 Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 /*
  * smartos-live: Path manipulation utility functions.
  */
-
-#include <libcmdutils.h>
+#include "custr.h"
 
 /*
  * Append a path string, "inp", to the end of the path string already present
diff --git a/src/common/strings/strpath.h b/src/common/strings/strpath.h
index 4e1cbc9c..ab76b80b 100644
--- a/src/common/strings/strpath.h
+++ b/src/common/strings/strpath.h
@@ -10,7 +10,7 @@
  */
 
 /*
- * Copyright 2015 Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 #ifndef _STRPATH_H
@@ -20,7 +20,7 @@
  * smartos-live: Path manipulation utility functions.
  */
 
-#include "libcmdutils.h"
+#include "custr.h"
 
 #ifdef __cplusplus
 extern "C" {
diff --git a/src/common/strings/tests/testpath.c b/src/common/strings/tests/testpath.c
index 37933219..634ffb84 100644
--- a/src/common/strings/tests/testpath.c
+++ b/src/common/strings/tests/testpath.c
@@ -10,7 +10,7 @@
  */
 
 /*
- * Copyright 2015 Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 #include <stdio.h>
@@ -20,8 +20,8 @@
 #include <err.h>
 #include <errno.h>
 #include <sys/debug.h>
-#include <libcmdutils.h>
 
+#include "custr.h"
 #include "strpath.h"
 
 typedef struct {
diff --git a/src/dockerinit/Makefile b/src/dockerinit/Makefile
index 8d7639c8..4c860c1a 100644
--- a/src/dockerinit/Makefile
+++ b/src/dockerinit/Makefile
@@ -10,7 +10,7 @@
 #
 
 #
-# Copyright 2015 Joyent, Inc.
+# Copyright 2017 Joyent, Inc.
 #
 
 BASE :=		$(PWD)
@@ -28,12 +28,12 @@ GXX =		$(NATIVEDIR)/usr/bin/g++
 SYSINCDIRS =	/usr/include
 SYSLIBDIRS =	/usr/lib /lib
 
-CPPFLAGS =	$(SYSINCDIRS:%=-isystem $(DESTDIR)/%) -D_LARGEFILE64_SOURCE
+CPPFLAGS =	$(SYSINCDIRS:%=-isystem $(DESTDIR)/%)
 LDFLAGS =	$(SYSLIBDIRS:%=-L$(DESTDIR)/%) \
 		-Wl,-zassert-deflib -Wl,-zfatal-warnings
 
 JSON_OBJS =	json-nvlist.o
-JSON_LIBS =	-lnvpair -lcmdutils
+JSON_LIBS =	-lnvpair
 
 MDATA_OBJS = \
 		mdata-client/dynstr.o \
@@ -48,7 +48,7 @@ MDATA_LIBS =	-lnsl -lsocket -lsmbios
 
 NET_LIBS =	-lipadm -linetutil -lnsl -lsocket
 
-STRINGS_OBJS =	strlist.o strpath.o
+STRINGS_OBJS =	custr.o strlist.o strpath.o
 
 DOCKER_OBJS =	docker-common.o
 
diff --git a/src/dockerinit/json-nvlist/json-nvlist.c b/src/dockerinit/json-nvlist/json-nvlist.c
index d686d77f..2a337365 100644
--- a/src/dockerinit/json-nvlist/json-nvlist.c
+++ b/src/dockerinit/json-nvlist/json-nvlist.c
@@ -10,7 +10,7 @@
  */
 
 /*
- * Copyright 2015 Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 #include <stdio.h>
@@ -21,6 +21,7 @@
 #include <libnvpair.h>
 #include <sys/ccompile.h>
 
+#include "custr.h"
 #include "json-nvlist.h"
 
 typedef enum json_type {
diff --git a/src/dockerinit/json-nvlist/json-nvlist.h b/src/dockerinit/json-nvlist/json-nvlist.h
index d7f81065..06b35144 100644
--- a/src/dockerinit/json-nvlist/json-nvlist.h
+++ b/src/dockerinit/json-nvlist/json-nvlist.h
@@ -10,14 +10,13 @@
  */
 
 /*
- * Copyright 2015 Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 #ifndef _JSON_NVLIST_H
 #define _JSON_NVLIST_H
 
 #include <libnvpair.h>
-#include <libcmdutils.h>
 
 #ifdef __cplusplus
 extern "C" {
diff --git a/src/dockerinit/src/docker-common.h b/src/dockerinit/src/docker-common.h
index 9085fdfb..07ec4ad1 100644
--- a/src/dockerinit/src/docker-common.h
+++ b/src/dockerinit/src/docker-common.h
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 /*
@@ -29,7 +29,7 @@
 #define DOCKER_COMMON_H
 
 #include <libnvpair.h>
-#include <libcmdutils.h>
+#include "custr.h"
 #include "../json-nvlist/json-nvlist.h"
 #include "strlist.h"
 
diff --git a/src/dockerinit/src/dockerinit.c b/src/dockerinit/src/dockerinit.c
index 96492682..2dfbd7a7 100644
--- a/src/dockerinit/src/dockerinit.c
+++ b/src/dockerinit/src/dockerinit.c
@@ -28,6 +28,7 @@
  * problems.
  */
 
+#include <dirent.h>
 #include <door.h>
 #include <errno.h>
 #include <fcntl.h>
diff --git a/src/routeinfo/Makefile b/src/routeinfo/Makefile
index ac599cf0..4aee0eee 100644
--- a/src/routeinfo/Makefile
+++ b/src/routeinfo/Makefile
@@ -10,7 +10,7 @@
 #
 
 #
-# Copyright 2015 Joyent, Inc.
+# Copyright 2017 Joyent, Inc.
 #
 
 BASE :=		$(PWD)
@@ -32,6 +32,7 @@ CPPFLAGS =	$(SYSINCDIRS:%=-isystem $(DESTDIR)/%)
 LDFLAGS =	$(SYSLIBDIRS:%=-L$(DESTDIR)/%) \
 		-Wl,-zassert-deflib -Wl,-zfatal-warnings
 
+CUSTR_DIR =	$(BASE)/../common/strings
 DOCKER_DIR =	$(BASE)/../dockerinit
 
 ROUTEINFO_OBJECTS = \
@@ -47,16 +48,18 @@ MDATA_OBJECTS = \
 		sunos.o \
 		unix_common.o
 
+custr.o :	CFLAGS += -I$(CUSTR_DIR)
 mdata_%.o :	CFLAGS += -I$(DOCKER_DIR)/mdata-client -D__HAVE_BOOLEAN_T
 
 OBJECTS = \
 		$(MDATA_OBJECTS:%=mdata_%) \
+		custr.o \
 		json-nvlist.o \
 		main.o
 
-json-nvlist.o :	CFLAGS += -I$(DOCKER_DIR)/json-nvlist
+json-nvlist.o :	CFLAGS += -I$(DOCKER_DIR)/json-nvlist -I$(CUSTR_DIR)
 
-LIBS =		-lnvpair -lnsl -lsocket -lsmbios -lcmdutils
+LIBS =		-lnvpair -lnsl -lsocket -lsmbios
 
 CTFCONVERT =	/usr/bin/true
 CTFMERGE =	/usr/bin/true
@@ -93,6 +96,10 @@ $(PROGRAM): $(OBJECTS)
 	$(CC) $(CPPFLAGS) $(LDFLAGS) $(CFLAGS) -o $@ $(OBJECTS) $(LIBS)
 	$(CTFMERGE) -L VERSION -o $@ $(OBJECTS)
 
+custr.o: $(CUSTR_DIR)/custr.c
+	$(COMPILE_C)
+	$(CTFCONVERT) -L VERSION $@
+
 mdata_%.o: $(DOCKER_DIR)/mdata-client/%.c
 	$(COMPILE_C)
 	$(CTFCONVERT) -L VERSION $@
diff --git a/src/routeinfo/main.c b/src/routeinfo/main.c
index fa9b6a2a..fa833eb7 100644
--- a/src/routeinfo/main.c
+++ b/src/routeinfo/main.c
@@ -10,7 +10,7 @@
  */
 
 /*
- * Copyright 2015 Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 /*
@@ -36,7 +36,6 @@
 #include <err.h>
 
 #include <libnvpair.h>
-#include <libcmdutils.h>
 
 #include <json-nvlist/json-nvlist.h>
 #include <mdata-client/common.h>
diff --git a/tools/tzcheck/main.c b/tools/tzcheck/main.c
index 503bf19a..922716a0 100644
--- a/tools/tzcheck/main.c
+++ b/tools/tzcheck/main.c
@@ -35,6 +35,8 @@
 #include <dirent.h>
 #include <ctype.h>
 #include <sys/avl.h>
+#include <limits.h>
+#include <unistd.h>
 
 #include "common.h"
 #include "parser.h"
-- 
2.21.0

